source/sn_coap_protocol.c

```diff
@@ -690,6 +692,8 @@ sn_coap_hdr_s *sn_coap_protocol_parse(struct coap_s *handle, sn_nsdl_addr_s *src
                     tr_debug("sn_coap_protocol_parse - send ack for duplicate message");
                     handle->sn_coap_tx_callback(response->packet_ptr,
                             response->packet_len, response->address, response->param);
+                } else {
+                    tr_error("sn_coap_protocol_parse - response not yet build");
```

Comment:
Is this a programming error, or does it happen currently?

Reply:
Well, it should not happen anymore, but at least this trace would inform us if there is still some unexpected corner-case.

Without this fix the code was going that path.